### ISSUE TITLE
The multi-view fitting code is implemented from which paper?

### ISSUE THREAD

**Issue by syguan96** [zju3dv/EasyMocap#158]

**Comment by pablovela5620** [comment]
In the acknowledgements section they specify
```
method for fitting 3D skeleton and SMPL model is similar to [TotalCapture](http://www.cs.cmu.edu/~hanbyulj/totalcapture/), without using point clouds.
```

I did want to also know why the Optimizer used in their code is the LBFGS while the total capture (and monocular version) use the Levenberg-Mardquat optimizer. 

Is this mostly because Ceres is used there (which has a native LM implementation) while Pytorch does not?

**Comment by chingswy** [comment]
We implement this following [smplify-x](https://github.com/vchoutas/smplify-x).
PyTorch is more friendly for common users. We also implement our code with Ceres but it's hard to compile and use it everywhere.

**Comment by chingswy** [comment]
Indeed, you can implement this with PyTorch. This can also be real-time. The key idea is calculate the Jacobian matrix manually and update the parameters with Gauss-Newton method.

**Comment by pablovela5620** [comment]
I did also find this recent library from meta, that implements Levenberg-Mardquat and other nonlinear optimizers
https://github.com/facebookresearch/theseus
https://sites.google.com/view/theseus-ai

Wanted to share if anyone is interested. Still looking into potentially using it to see if convergence speed improves v.s. LBFGS or something like Adam

**Comment by JACKYLUO1991** [comment]
@pablovela5620 Hi, I am also looking into how to optimize it in real time, can you provide some code snippets or ideas?

**Comment by pablovela5620** [comment]
@JACKYLUO1991 I mostly focus on hand pose estimation, but I would recommend looking into this paper as to how to potentially make things realtime. 

https://research.facebook.com/publications/megatrack-monochrome-egocentric-articulated-hand-tracking-for-virtual-reality/

It can probably be modified to other modalities such as body pose estimation, but this is what I have in mind

**Comment by JACKYLUO1991** [comment]
@pablovela5620 It may not have been clear, what I was trying to understand was how LM accelerates optimisation.

**Comment by pablovela5620** [comment]
This link may be helpful
https://docs.mantidproject.org/v3.7.2/concepts/FittingMinimizers.html

Since Levenberg Mardquat is basically a mixture between Gauss-Newton and Gradient Descent, it's both fast and robust. Supposedly it's more accurate than LBFGS and faster for this type of problem. But I haven't implemented yet, in the process of doing so
